pkg/webui/components/form/index.js

```diff
@@ -23,6 +24,30 @@ import FormField from './field'
 import FormSubmit from './submit'
 
 class InnerForm extends React.PureComponent {
+  constructor (props) {
+    super(props)
+    this.focusGroup = React.createRef()
+  }
+
+  componentDidUpdate (prevProps) {
+    const { formError, isSubmitting, isValid } = this.props
+    const { isSubmitting: prevIsSubmitting, formError: prevFormError } = prevProps
+
+    // Scroll form notification into view if needed
+    if (formError && !prevFormError) {
+      scrollIntoView(this.focusGroup.current, { behavior: 'smooth' })
+    }
+
+    // Scroll invalid fields into view if needed and focus them
+    if (!prevIsSubmitting && !formError && isSubmitting && !isValid) {
```

Comment:
why not
```suggestion
    if (!prevIsSubmitting && isSubmitting && !isValid) {
```
imagine this:
1. the user submits the form and gets some error from the backend
2. the user changes data in the form
3. submits
4. gets frontend validation error

The `formError` is still rendered since the `onSubmit` function wasnt called, however I would still expect the invalid field being focused.